src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin.StrongNamed/OmniSharpWorkspaceProjectStateChangeDetector.cs

```diff
@@ -45,9 +44,12 @@ private class ProjectSnapshotManagerWorkspaceProjectStateChangeDetector : Worksp
             // thread. OmniSharp currently has an issue where they update the Solution on multiple different threads resulting
             // in change events dispatching through the Workspace on multiple different threads. This normalizes
             // that abnormality.
-#pragma warning disable VSTHRD100 // Avoid async void methods
-            protected override async void InitializeSolution(Solution solution)
-#pragma warning restore VSTHRD100 // Avoid async void methods
+            protected override void InitializeSolution(Solution solution)
+            {
+                _ = InitializeSolutionAsync(solution, CancellationToken.None);
+            }
+
+            private async Task InitializeSolutionAsync(Solution solution, CancellationToken cancellationToken)
```

Comment:
This should have `try...catch` in it, right? or Does RunOnDispatcherThread do that?

Reply:
There is a try...catch inside the Dispatcher call, or do you mean that it should be at some other level?

Reply:
So I could be wrong, but I thought when changing from `async void` to `_ = SomeAsync()` it was important to make sure the method being called had a try catch in it to make sure exceptions aren't lost. This doesn't follow that pattern, so just want to make sure its not something we need to worry about.

Reply:
Oh duh, you're not saying that it LACKS try-catch, you're saying that the try-catch does not capture the entire possible failure-space. Please excuse my Friday-brain, will expand.